packages/react/src/components/Storage/FileUploader/types.ts

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { DragActionHandlers } from './hooks/useFileUploader/types';
+import { FileName } from '@aws-amplify/ui';
+
+export type SetShowPreviewer = (show: boolean) => void;
+type LevelInfo = 'public' | 'protected' | 'private';
+export type FileTypes = File[];
+
+export interface UploaderButtonProps {
+  acceptedFileTypes: Array<string>;
+  multiple?: boolean;
+  onUpload: (event: React.ChangeEvent<HTMLInputElement>) => void;
```

Comment:
Having a hard time understanding why we need an `onUpload` prop when we could just use the idiomatic `onClick`?

Reply:
I'l change it, I'll revisit it in the next PR